movie-browser/src/App.js

```diff
@@ -1,23 +1,24 @@
 import React, { Component } from 'react';
-import Header from './components/Header.js';
-import Featured from './components/Featured.js';
-import Movies from './components/Movies.js';
-import Footer from './components/Footer.js';
-import MostViewed from './components/MostViewed.js';
-//import fetchMovies from './components/Api.js';
+import Header from './components/Header';
+import Featured from './components/Featured';
+import Movies from './components/Movies';
+import Footer from './components/Footer';
+import MostViewed from './components/MostViewed';
+import fetchMovies from './Api';
 
 class App extends Component {
   state = { 
     movies: [],
-    shouldShowAllMovies: false
+    shouldShowAllMovies: false,
+    isLoading: false
    };
 
   componentDidMount() {
-    fetch('https://ghibliapi.herokuapp.com/films')
-    .then(response => response.json())
-    .then(data => this.setState({ movies: data }));
-  }
-//    fetchMovies('films');
+    this.setState({ isLoading: true });
```

Comment:
if the initial state is `isLoading: true` move that to the state itself. Remove this line.

Reply:
The initial state is false. This is the starting state in app.js:
```
  state = { 
    movies: [],
    shouldShowAllMovies: false,
    isLoading: false
   };
```
The `isLoading` state only changes inside `componentDidMount` to true. And the false once the API fetch is complete.